pkg/db-repo/repository.go

```diff
@@ -217,6 +217,16 @@ func (r *repository) Delete(ctx context.Context, value ModelBased) error {
 	return err
 }
 
+func (r *repository) isQueryableModel(model interface{}) bool {
+	modelType := reflect.TypeOf(model)
```

Comment:
Solving this via reflection requires the model to always match the name on the metadata. I wouldn't rely on this assumption.

Can you find a way to e.g. compare the table names in the from clause?
Maybe the ORM provides a way to get the table name from the model

Reply:
Applied